File: backend/src/SiteYonetimi.Infrastructure/Entities/Shared/Payment.cs
using SiteYonetimi.Shared.Enums;

namespace SiteYonetimi.Infrastructure.Entities.Shared;

public class Payment
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SiteId { get; set; }
    public Guid UnitId { get; set; }
    public decimal Amount { get; set; }
    public DateTime DueDate { get; set; }
    public DateTime? PaidDate { get; set; }
    public PaymentStatus Status { get; set; } = PaymentStatus.Pending;
    public string? Description { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Unit Unit { get; set; } = null!;
    public ICollection<PaymentItem> Items { get; set; } = new List<PaymentItem>();
}

File: frontend/src/types/payment.ts
export type PaymentStatus = 0 | 1 | 2  // Pending=0, Paid=1, Overdue=2

export const PaymentStatusLabel: Record<PaymentStatus, string> = {
  0: 'Beklemede',
  1: 'Ödendi',
  2: 'Gecikmiş',
}

export const PaymentStatusColor: Record<PaymentStatus, string> = {
  0: 'bg-yellow-100 text-yellow-800',
  1: 'bg-green-100 text-green-800',
  2: 'bg-red-100 text-red-800',
}

export interface PaymentItemDto {
  name: string
  amount: number
}

export interface PaymentSummaryDto {
  id: string
  siteId: string
  unitId: string
  unitDoorNumber?: string
  amount: number
  items: PaymentItemDto[]
  dueDate: string
  paidDate?: string
  status: PaymentStatus
  description?: string
  createdAt: string
}

export interface BulkCreatePaymentsDto {
  buildingId?: string
  items: PaymentItemDto[]
  dueDate: string
  description?: string
}

export interface CreatePaymentDto {
  unitId: string
  items: PaymentItemDto[]
  dueDate: string
  description?: string
}

export interface UpdatePaymentDto {
  items: PaymentItemDto[]
  dueDate: string
  description?: string
}

export interface UpdatePaymentStatusDto {
  status: PaymentStatus
  paidDate?: string
}

export interface ImportPreviewRow {
  buildingName: string
  doorNumber: string
  items: PaymentItemDto[]
  total: number
}

export interface ImportSkippedRow {
  buildingName: string
  doorNumber: string
  reason: string
}
